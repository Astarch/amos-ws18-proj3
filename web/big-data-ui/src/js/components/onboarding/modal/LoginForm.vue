<template>
  <div id="form-login" class="onboarding-form form-login" :class="{active: active}">
    <input id="form-username" type="text" name="user" placeholder="Username" v-model="username" v-on:input="hideLoginError" v-on:keyup.enter.stop="submit()" v-on:change="hideLoginError">
    <input id="form-password" type="password" name="password" placeholder="Password" v-model="password" v-on:keyup.enter.stop="submit()" v-on:input="hideLoginError" v-on:change="hideLoginError">
    <div v-if="hasError" class="alert alert-danger" role="alert">
      {{errorText}}
    </div>
    <input id="form-login-submit" type="submit" :class="{disabled: reqStatus.pending}" v-on:click.stop="submit()" value="Login">
    <div class="links">
      <a href="" v-on:click.prevent.stop="openRegistration()">Don't have an account yet?</a>
    </div>
  </div>
</template>

<script>
  import {
    mapState
  } from "vuex";
  import {
    isPasswordValid,
    isNameValid
  } from "../../../utils/validation";
  import RequestStatus from 'src/js/models/RequestStatus';
  import {
    ServerErrors
  } from 'src/js/utils/constants';
  
  export default {
    name: "login-form",
    props: {
      active: Boolean
    },
    data: () => ({
      username: "",
      password: "",
      errorText: "",
      hasError: false,
      reqStatus: new RequestStatus()
    }),
    computed: mapState({
      // arrow functions can make the code very succinct!
      user: state => state.user.user,
    }),
    mounted() {

      // fill in username if user just registered
      if (this.user != undefined && this.user.username != undefined && this.user.username.length > 2) {
        this.username = this.user.username
      }
    },
    methods: {
      submit() {
        var data = {
          form: this.type
        };
  
        let validLoginData = this.validateLogin(this.username, this.password);
        if (validLoginData) {
          data.password = this.password;
          data.user = this.username;
          this.doLogin(this.username, this.password);
        }
      },
      doLogin(name, password) {
        this.reqStatus.pending = true;
        this.$store.dispatch('loginUser', {
            username: name,
            password
          })
          .then(requestStatus => {
            this.reqStatus = requestStatus;
            console.log(this.user)
            this.$router.push({
              path: "engagement"
            });
          }).catch(requestStatus => {
            this.reqStatus = requestStatus;
            console.log(requestStatus)
            if (this.reqStatus !== ServerErrors.ERROR_SERVER) {
              this.showLoginError("Login failure - please try again!");
            } else {
              this.showLoginError("Server error - please try again later!");
            }
          })
        /*api.auth
          .postLogin(name, password)
          .then(response => {
            this.$router.push({
              path: "engagement"
            });
          })
          .catch(error => {
            let isServerError =
              !error ||
              !error.response ||
              !error.response.status ||
              error.response.status >= 500;
            if (!isServerError) {
              this.showLoginError("Login failure - please try again!");
            } else {
              this.showLoginError("Server error - please try again later!");
            }
          });*/
      },
      validateLogin(name, password) {
        let isValid = false;
        if (!isNameValid(name)) {
          this.showLoginError("Please enter a valid username!");
        } else if (!isPasswordValid(password)) {
          this.showLoginError("Please enter a valid password!");
        } else {
          isValid = true;
          this.hideLoginError();
        }
  
        return isValid;
      },
      showLoginError(message) {
        if (typeof message === "string" || message instanceof String) {
          this.hasError = true;
          this.errorText = message;
        } else {
          this.hasError = false;
        }
      },
      hideLoginError() {
        this.hasError = false;
      },
      openRegistration() {
        this.$emit("openRegistration");
      }
    }
  };
</script>

<style lang='scss'>
  @import "../../../../scss/forms";
</style>
