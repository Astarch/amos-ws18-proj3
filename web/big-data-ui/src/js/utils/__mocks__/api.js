/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/

const auth = {
  postLogin: jest.fn(() => Promise.resolve({success: true})),
  postRegistration: jest.fn(() => Promise.resolve({success: true, response:{status: 401}})),
  postCompleteRegistration: jest.fn(() => Promise.resolve({success: true})),
  doLogout: jest.fn(() => Promise.resolve({success: true}))
};

/***** user *****/

const user = {
  getAll: jest.fn(() => Promise.resolve({success: true}))
};

/***** graph *****/

const graph = {
  getData: jest.fn(() => Promise.resolve({success: true}))
};

const api = {
  auth: auth,
  user: user,
  graph: graph
}

export  {
  api
};

