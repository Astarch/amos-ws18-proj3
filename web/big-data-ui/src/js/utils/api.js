import axios from 'axios';
import qs from 'qs';
import store from 'store2'


/******************** Base API / Client Configuration ********************/

let port = "8081/"
let localhostUrl = "http://localhost:" + port;
let stagingUrl = "http://18.216.129.153:" + port;
let liveUrl = "http://18.216.122.218:" + port;
let baseUrl = stagingUrl
baseUrl = (GIT.BRANCH && !GIT.BRANCH.includes("master")) ? liveUrl : stagingUrl;



function writeAuthToken(token) {
    store.session('authToken', token)
    return token
}

function getAuthToken() {
    return store.session('authToken')
}


const axiosInstance = axios.create({
    baseURL: baseUrl,
});

if (getAuthToken() && getAuthToken().length > 5) {
    Object.assign(axiosInstance.defaults, { headers: { 'x-auth-token': getAuthToken() } });
}

// Intercepts all responses, retrieves the x auth token and sets it to all following requests headers!
axiosInstance.interceptors.response.use(response => {
    let authToken = getAuthToken()
    if (response.status == 401) {
        authToken = writeAuthToken("")
    } else if (response && response.headers && response.headers['x-auth-token']) {
        let authToken = writeAuthToken(response.headers['x-auth-token']);
        if (authToken && authToken.length > 5) {
            Object.assign(axiosInstance.defaults, { headers: { 'x-auth-token': authToken } });
        }
    }

    return response;
}, error => {
    // Do something with response error
    return Promise.reject(error);
});

export default axiosInstance;

export const isLoggedIn = () => {
    let authToken = getAuthToken() ? getAuthToken() : authToken;
    return authToken && authToken.length > 5
}

export const logoutUser = () => {
    writeAuthToken("")
}



/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => axiosInstance.post('/auth/login', qs.stringify({
    username: username,
    password: password
}));

const postCompleteRegistration = (username,
    password,
    firstname = username,
    lastname = username,
    email = username + "@test.com",
    address = "addres1",
    phone = "phoneNr") => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "firstname": firstname,
    "lastname": lastname,
    "email": email,
    "address": address,
    "phone": phone
});

const postRegistration = (username,
    email,
    password) => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "email": email,
});

const doLogout = () => axiosInstance.get('/auth/logout');

const auth = {
    postLogin,
    postRegistration,
    postCompleteRegistration,
    doLogout
};

/***** user *****/
const getAll = () => axiosInstance.get('/api/user/getAll');

const user = {
    getAll
};

/***** graph *****/
const getData = (path) => axiosInstance.get(path);

const graph = {
    getData
};

const api = {
    auth,
    user,
    graph
}

export {
    api
};