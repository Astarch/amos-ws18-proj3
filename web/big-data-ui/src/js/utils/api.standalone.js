import axios from 'axios';
import qs from 'qs';
import storage from 'store2'
import store from 'src/js/store';

import trump from 'src/assets/wordcounts/trump'
const availableTerms = ["trump"]



/******************** Base API / Client Configuration ********************/

function writeAuthToken(token) {
  storage.set('authToken', token)
  return token
}

function getAuthToken() {
  return storage.get('authToken')
}

export const isLoggedIn = () => {
  let authToken = getAuthToken() ? getAuthToken() : authToken;
  return authToken && authToken.length > 5
}

export const logoutUser = () => {
  writeAuthToken("")
}


/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => {
  return new Promise((resolve, reject) => {
    if (username == "user2" && password == "pass2") {
      const response = {
        data: {
          id: 123,
          username: "user2",
          email: "user2@test.de",
          firstname: "John",
          lastname: "Doee",
          authorities: [],
          isLoggedIn: true
        }
      }
      writeAuthToken("loggedIn")
      resolve(response)
    } else {
      const errorResponse = {
        status: 401
      }
      reject(errorResponse)
    }
  })
}
//axiosInstance.post('/auth/login', qs.stringify({
//    username: username,
//    password: password
//}));

const postCompleteRegistration = (username,
  password,
  firstname = username,
  lastname = username,
  email = username + "@test.com",
  address = "addres1",
  phone = "phoneNr") => {
  return new Promise((resolve, reject) => {
    const errorResponse = {
      status: 401
    }
    reject(errorResponse)
  })
}
//axiosInstance.post('/auth/register', {
//    "username": username,
//    "password": password,
//    "firstname": firstname,
//    "lastname": lastname,
//    "email": email,
//    "address": address,
//    "phone": phone
//});

const postRegistration = (username, firstname, lastname,
  email,
  password) => {
  return new Promise((resolve, reject) => {
    const errorResponse = {
      status: 401
    }
    reject(errorResponse)
  })
}
//axiosInstance.post('/auth/register', {
//    "username": username,
//    "firstname": firstname,
//    "lastname": lastname,
//    "password": password,
//    "email": email,
//});



const auth = {
  postLogin,
  postRegistration,
  postCompleteRegistration
};

const localFirebaseUrl = "http://localhost:5000/pretrendr-d5939/us-central1"
const remoteFirebaseUrl = "https://us-central1-pretrendr-d5939.cloudfunctions.net"
const firebaseUrl = remoteFirebaseUrl
/***** graph *****/
const getWordcountByDay = (query, method, timerange, normalize) => {
  /*return new Promise((resolve, reject)=> {
      if(query.indexOf("trump")>= 0){
          const response = {
              data: trump
          }
          resolve(response)
      }else{
          const errorResponse = {
              status: 404
          }
          reject(errorResponse)
      }
  })*/
  return axios.get(`${firebaseUrl}/getWordcountData?query=${query}&method=${method}`)

  //axiosInstance.get(`/api/gdelt/wordcountByDay/${query}?${timerange}&normalize=${normalize? 'true': 'false'}&method=${method}`);
}
const uploadDataToFirestore = (query, method, timerange, data) => {
  const uploadData = {
    query,
    method,
    timerange,
    data,
  };
  return axios.post(`${firebaseUrl}/addWordcountData`, uploadData)    
}

const getCachedTrends = () => axios.get(`${firebaseUrl}/getAllCachedTerms`)


const graph = {
  getWordcountByDay,
  getCachedTrends
};

const api = {
  auth,
  graph
}

export {
  api
};
