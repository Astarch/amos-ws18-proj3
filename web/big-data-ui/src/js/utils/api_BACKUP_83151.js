import realClient, {api as realApi, logoutUser as realLogoutUser, isLoggedIn as realIsLoggedIn}  from './api.real'
import standaloneClient, {api as standaloneApi, logoutUser as standaloneLogoutUser, isLoggedIn as standaloneIsLoggedIn}  from './api.standalone'
console.dir(standaloneApi)

const isStandaloneClient = false

if(isStandaloneClient){
    //export default standaloneClient;
}else{
    //export default realClient;
}

<<<<<<< HEAD
export const logoutUser = () => {
    writeAuthToken("")
}



/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => axiosInstance.post('/auth/login', qs.stringify({
    username: username,
    password: password
}));

const postCompleteRegistration = (username,
    password,
    firstname = username,
    lastname = username,
    email = username + "@test.com",
    address = "addres1",
    phone = "phoneNr") => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "firstname": firstname,
    "lastname": lastname,
    "email": email,
    "address": address,
    "phone": phone
});

const postRegistration = (username, firstname,lastname,
    email,
    password) => axiosInstance.post('/auth/register', {
    "username": username,
    "firstname": firstname,
    "lastname": lastname,
    "password": password,
    "email": email,
});

const doLogout = () => axiosInstance.get('/auth/logout');

const auth = {
    postLogin,
    postRegistration,
    postCompleteRegistration,
    doLogout
};

/***** user *****/
const getAll = () => axiosInstance.get('/api/user/getAll');

const user = {
    getAll
};

/***** graph *****/
const getWordcountByDay = (query, method, timerange, normalize) =>
    axiosInstance.get(`/api/gdelt/wordcountByDay/${query}?${timerange}&normalize=${normalize? 'true': 'false'}&method=${method}`);

const graph = {
    getWordcountByDay
};

const uploadDataToFirestore = (query, method, timerange, data) => {
    const uploadData = {
      query,
      method,
      timerange,
      data,
    };
    return axios.post('https://us-central1-pretrendr-d5939.cloudfunctions.net/addWordcountData', uploadData)    
  }

const api = {
    auth,
    user,
    graph,
    uploadDataToFirestore
}
=======
const isLoggedIn = isStandaloneClient? standaloneIsLoggedIn : realIsLoggedIn
const logoutUser = isStandaloneClient? standaloneLogoutUser : realLogoutUser
const api = isStandaloneClient? standaloneApi : realApi
>>>>>>> initial commit to use frontend without backend

export {
    isLoggedIn,
    logoutUser,
    api
}
