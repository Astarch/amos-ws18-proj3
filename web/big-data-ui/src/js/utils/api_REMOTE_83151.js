import realClient, {api as realApi, logoutUser as realLogoutUser, isLoggedIn as realIsLoggedIn}  from './api.real'
import standaloneClient, {api as standaloneApi, logoutUser as standaloneLogoutUser, isLoggedIn as standaloneIsLoggedIn}  from './api.standalone'
console.dir(standaloneApi)

const isStandaloneClient = false

if(isStandaloneClient){
    //export default standaloneClient;
}else{
    //export default realClient;
}

const isLoggedIn = isStandaloneClient? standaloneIsLoggedIn : realIsLoggedIn
const logoutUser = isStandaloneClient? standaloneLogoutUser : realLogoutUser
const api = isStandaloneClient? standaloneApi : realApi

export {
    isLoggedIn,
    logoutUser,
    api
}
