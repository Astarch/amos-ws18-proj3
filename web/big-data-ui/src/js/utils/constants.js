const FormTypeEnum = { "login": 1, "register": 2 };
Object.freeze(FormTypeEnum);

const ERROR_SERVER = 'ERROR_SERVER' // -> 500 status code
const ERROR_AUTHENTICATION = 'ERROR_AUTHENTICATION' // -> 401 status code
const ERROR_PERMISSION = 'ERROR_PERMISSION' // -> 403 status code
const ERROR_REQUEST = 'ERROR_REQUEST' // -> 404 status code / everything else
const ERROR_NONE = 'ERROR_NONE' // -> 404 status code / everything else

const ServerErrors = {
    ERROR_AUTHENTICATION,
    ERROR_SERVER,
    ERROR_PERMISSION,
    ERROR_REQUEST,
    ERROR_NONE
}

export {
    FormTypeEnum,
    ServerErrors
}