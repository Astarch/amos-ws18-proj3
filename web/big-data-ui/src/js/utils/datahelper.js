function normalizeData(graphdata) {
    //console.time("normalizeData")
    let max = Object.values(graphdata).reduce((max, current) => {
        return current > max ? current : max
    }, -1)
    let normalizeFactor = 1 / (max / 100)
    let normalized = Object.keys(graphdata).reduce((previous, current) => {
        previous[current] = Math.round(graphdata[current] * normalizeFactor)
        return previous;
    }, {});
    //console.timeEnd("normalizeData")
    return normalized
}

function normalizeDataArray(dataArray) {
    let normalizedArray = dataArray.map(value => {
        return Object.assign({}, value, { data: normalizeData(value.data) })
    })
    return normalizedArray
}

function median(values) {
    values = values.slice(0).sort(function(a, b) { return a - b; });

    return middle(values);
}

function middle(values) {
    var len = values.length;
    var half = Math.floor(len / 2);

    if (len % 2)
        return (values[half - 1] + values[half]) / 2.0;
    else
        return values[half];
}

function calculateMinMaxAvgMedianOfData(graphdata) {
    const values = Object.values(graphdata)
    let min = parseInt(Math.min(...values), 10);
    let max = parseInt(Math.max(...values), 10);
    let sum = 0;
    for (var i = 0; i < values.length; i++) {
        sum += parseInt(values[i], 10); //don't forget to add the base
    }
    let avg = parseInt(sum / values.length, 10);
    let _median = parseInt(median(values), 10);

    return Object.assign({}, {
        min,
        max,
        avg,
        median: _median
    })
}

function aggregateDataArray(dataArray) {
    let additionalDataArray = dataArray.map(value => {
        return Object.assign({}, value, calculateMinMaxAvgMedianOfData(value.data))
    })
    return additionalDataArray
}



export {
    normalizeDataArray,
    aggregateDataArray
}