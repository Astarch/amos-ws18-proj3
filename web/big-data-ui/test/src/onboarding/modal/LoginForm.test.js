import { shallow, mount } from 'vue-test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import LoginForm from 'src/js/components/onboarding/modal/LoginForm'
import { api } from 'src/js/utils/api'
jest.mock('src/js/utils/api');

Vue.use(Vuex)

describe('LoginForm.test.js', () => {

    let store
    let actions
    let state

    beforeEach(() => {
        jest.resetModules()
        jest.clearAllMocks()
        state = {
            user: {
                user: {
                    id: '',
                    username: '',
                    email: '',
                    encryptedPassword: '',
                    firstname: '',
                    lastname: '',
                    authorities: []
                },
                loginStatus: {
                    pending: false,
                    success: false,
                    failure: false,
                    failureType: ''
                }
            }
        }
        actions = {
            loginUser: jest.fn()
        }
        store = new Vuex.Store({
            state,
            actions
        })
    })

    test('has the expected html structure when not active', () => {
        const cmp = shallow(LoginForm, { store })
        cmp.setProps({ active: false })
        expect(cmp.element).toMatchSnapshot('inactive')
    })

    test('has the expected html structure when active', () => {
        const cmp = shallow(LoginForm, { store })
        cmp.setProps({ active: true })
        expect(cmp.element).toMatchSnapshot('active')
    })

    test('to register clicked event is emitted', () => {
        const cmp = shallow(LoginForm, { store })
        cmp.find('div.links > a').trigger('click')
        expect(cmp.emitted().openRegistration).toBeTruthy()
    })


    test('error is shown when submitting empty inputs', () => {
        const cmp = shallow(LoginForm, { store })
        cmp.find('#form-login-submit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeTruthy()
    })

    test('error is shown when submitting empty password', () => {
        const cmp = shallow(LoginForm, { store })
        cmp.find('input#form-username').element.value = "test"
        cmp.find('input#form-username').trigger('input')
        cmp.find('#form-login-submit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeTruthy()
    })

    test('error is shown when submitting empty username', () => {
        const cmp = shallow(LoginForm, { store })
        cmp.find('input#form-password').element.value = "test"
        cmp.find('input#form-password').trigger('input')
        cmp.find('#form-login-submit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeTruthy()
    })

    test('correct data is submitted', () => {
        const cmp = shallow(LoginForm, { store })
        cmp.find('input#form-password').element.value = "test1234"
        cmp.find('input#form-password').trigger('input')
        cmp.find('input#form-username').element.value = "test"
        cmp.find('input#form-username').trigger('input')
        cmp.find('#form-login-submit').trigger('click')
        let errorAlert = cmp.find('.alert')
        expect(errorAlert.exists()).toBeFalsy()

        expect(actions.loginUser.mock.calls).toHaveLength(1)
    })

})