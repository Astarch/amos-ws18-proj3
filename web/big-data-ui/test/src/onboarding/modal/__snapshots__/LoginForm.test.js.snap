// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm.test.js has the expected html structure when active: active 1`] = `
<div
  class="onboarding-form form-login active"
  id="form-login"
>
  <input
    id="form-username"
    name="user"
    placeholder="Username"
    type="text"
  />
   
  <input
    id="form-password"
    name="password"
    placeholder="Password"
    type="password"
  />
   
  <!---->
   
  <input
    class=""
    id="form-login-submit"
    type="submit"
    value="Login"
  />
   
  <div
    class="links"
  >
    <a
      href=""
    >
      Don't have an account yet?
    </a>
  </div>
</div>
`;

exports[`LoginForm.test.js has the expected html structure when not active: inactive 1`] = `
<div
  class="onboarding-form form-login"
  id="form-login"
>
  <input
    id="form-username"
    name="user"
    placeholder="Username"
    type="text"
  />
   
  <input
    id="form-password"
    name="password"
    placeholder="Password"
    type="password"
  />
   
  <!---->
   
  <input
    class=""
    id="form-login-submit"
    type="submit"
    value="Login"
  />
   
  <div
    class="links"
  >
    <a
      href=""
    >
      Don't have an account yet?
    </a>
  </div>
</div>
`;
